uwblueprint/headsupguys
# Auth Flow Screens + Cognito

## Notion ticket link

[Cognito Account Creation](https://www.notion.so/uwblueprintexecs/Cognito-Link-Create-Account-Email-Password-cd0b121a92424b6c8274acb1985ef940)
[Cognito Reset Password](https://www.notion.so/uwblueprintexecs/Cognito-Link-Reset-Password-68373af6636b47ac9525c81095635725)
[Cognito Account Google](https://www.notion.so/uwblueprintexecs/Create-Account-Google-9b02e41ab4814802a2a7595cbf347c2c)



## Implementation description
* amplify folder stores all the configurations for amplify for Cognito setup
* Signup, Login, Reset pages are all designed with stages in mind, each individual screen on the page is considered a stage and you progress to the next stage on button click, which is disabled unless the input information passes the requirements for each stage
* Auth functionality is used from Amplify which allows for the connection to Cognito to be established 



## Steps to test
1. Install the Amplify CLI, [instructions here](https://docs.amplify.aws/cli/start/install) and set up with hug aws account credentials
2. Once the Amplify CLI is installed run `amplify pull --appId d1u33y46xg7v0p --envName dev` to pull the Cognito backend setup locally, this will make sure Amplify is connecting to the correct user pool etc
3. `/signup` `/login` `/reset` are the 3 main pages with the Cognito functionality and screen designs for Auth
4. Account already exists collision checking should occur and validation on passwords to meet our requirements. 



## What should reviewers focus on?
*  Look and feel of the screens on signup, login, reset
* Functionality with Cognito and handling signup, login, and reset password scenarios


## Checklist
